### ISSUE TITLE
Automatic Container Detection fails on iOS

### ISSUE THREAD

**Issue by dansiegel** [dansiegel/Prism.Container.Extensions#66]
# Description

Part of the new feature set introduced as part of issue #60 was that we will automatically detect the Container so as to prevent the need to produce a completely separate package just for a container implementation on a specific platform. While this is great and works fantastic for the unit tests, it fails miserably on iOS where because there is no strong reference in code to the container implementation's assembly.

## Solution

In order to ensure that the assembly is loaded in the AppDomain, we need to add:

```cs
var _ = PrismContainerExtensions.Current;
```

This of course is pretty ugly. An alternative is to add:

```cs
[assembly: Preserve(typeof(PrismContainerExtension))]
```

In order to make this easier to deal with a targets file should be bundled with the Container NuGet that will add a file to be compiled with the Preserve attribute.

**Comment by pdonovan** [comment]
Either I'm not understanding this fix, or it's not actually fixed. It works when building for an iOS Simulator, but not for a real iOS device. Both are Debug builds.

If I add

`var _ = PrismContainerExtensions.Current;`

to the AppDelegate.Finished Launching() method then it works. If I don't, I get:

System.DllNotFoundException: 'No assembly containing a reference to an IContainerExtension implementation could be found. You must have a reference to one of the DI Container Extensions in your final project'
from public App(IPlatformInitializer initializer) (or its base class)


Using version 7.2.0.700 of Prism.Forms.Extended, Prism.Container.Extensions and Prism.DryIoc.Extensions.

Both my PrismTest and PrismTest.iOS projects have references to those 3 Nuget packages. The PrismTest.Android project works OK.

If I try with Prism.Microsoft.DependencyInjection.Extensions then it gets further, but results in:
System.InvalidOperationException: Unable to resolve service for type 'Prism.Navigation.INavigationService' while attempting to activate 'PrismTest.ViewModels.MainPageViewModel'

but I think that's an issue for a different ticket...

**Comment by dansiegel** [comment]
@pdonovan thanks for flagging. I'm reopening this issue to take a look at.

**Comment by dansiegel** [comment]
I should add please be sure to try the latest build.

**Comment by pdonovan** [comment]
My comment had a typo - I am already using 7.2.0.799, not .700.

**Comment by dansiegel** [comment]
this should be resolved now. Unfortunately it appears that the PreserveAttribute doesn't preserve as well as it's supposed to... but the important thing is that it's fixed and the latest build 7.2.0.844 should work for you.

**Comment by pdonovan** [comment]
Great! I may not be able to test it here until after the holidays, but I will let you know if I still have issues.

**Comment by pdonovan** [comment]
Hi @dansiegel.
I've tested with the latest version (844) of all the extension packages and it's still not working. I can't see what I'm doing wrong. On both iOS and Android, the app stops with `Exception - High: System.InvalidOperationException: Unable to resolve service for type 'Prism.Navigation.INavigationService' while attempting to activate 'PrismTest.ViewModels.MainPageViewModel'.`

I'm using MEDI rather than DryIoc, and not using Shiny, but other than that I really can't see a difference between my test and the sample at https://github.com/dansiegel/Prism.Container.Extensions/tree/master/sample/PrismSample

I have attached a zip of my test project:
[PrismTest.zip](https://github.com/dansiegel/Prism.Container.Extensions/files/4028252/PrismTest.zip)
